core/App/ehsm_napi.cpp

```diff
@@ -36,53 +36,37 @@
 using namespace std;
 using namespace EHsmProvider;
 
-static char* StringToChar(string str)
-{
-    char *retChar = NULL;
-    if (str.size() > 0) {
-        int len = str.size();
-        retChar = (char *)malloc(len * sizeof(uint8_t));
-        if(retChar != nullptr){
-            memset(retChar, 0, len);
-            memcpy(retChar, str.c_str(), len);
-        }
+
+extern "C" {
+
+/*
+create the enclave
+@return
+[string] json string
+    {
+        code: int,
+        message: string,
+        result: {}
+    }
+*/
+char* NAPI_Initialize(){
+    RetJsonObj retJsonObj;
+    ehsm_status_t ret = EH_OK;
+    
+    ret = Initialize();
+    if (ret != EH_OK) {
+        retJsonObj.setCode(retJsonObj.CODE_FAILED);
+        retJsonObj.setMessage("Server exception.");
     }
-    return retChar;
+    return retJsonObj.toChar();
 }
-typedef struct {
-    const int CODE_SUCCESS = 200;
-    const int CODE_FAILED = 500;
-    int code = CODE_SUCCESS;
-    std::string msg = "success!";
-    std::string jsonStr;
-	
-	void setCode(int code){code = code;};
-	void setMessage(string message){msg = message;};
-	void addData(string key, uint32_t data) {
-        if(jsonStr.size() > 0){
-            jsonStr += ",";
-        }
-        jsonStr += "\""+key+"\" : " + "\""+std::to_string(data)+"\"";
-    };
-	void addData(string key, string data) {
-        if(jsonStr.size() > 0){
-            jsonStr += ",";
-        }
-        jsonStr += "\""+key+"\" : " + "\""+data+"\"";
-    };
-
-    char* toChar() {
-        std::string retString = "{";
-        retString += "\"code\":" + std::to_string(code);
-        retString += ",\"message\":\"" + msg;
-        retString += "\"";
-        retString += ",\"result\":{"+jsonStr+"}";
-        retString += "}";
-		return StringToChar(retString);
-	};
-} RetJsonObj;
 
-extern "C" {
+/*
+destory the enclave
+*/
+void NAPI_Finalize(){
```

Comment:
not found the correct place to call?

Reply:
This method is called in nodejs.

Reply:
I only find ehsm_napi.NAPI_Initialize() is called.   Where is NAPI_Finalize()?

Reply:
It will be fix.